File: programs-basic-java/src/test/java/com/local/numbertypes/DuckNumberTest.java
package com.local.numbertypes;

import static com.local.numbertypes.NumberType.DUCK_NUMBER;

import java.util.List;

import org.junit.jupiter.api.Test;

class DuckNumberTest extends AbstractNumberTest {

	@Test
	void testDuckNumbers() {
		testNumbersTypeByFile(DUCK_NUMBER, "ducknumbers.txt");
	}

	@Test
	void testDuckNumbersByList() {
		// Pass the order as 5
		testNumbersTypeByList(DUCK_NUMBER, List.of(10, 20, 30, 40, 50, 60, 70, 80, 01111111), 8);
	}
}

File: programs-basic-kotlin/src/test/kotlin/com/local/numbertypes/DuckNumberTest.kt
package com.local.numbertypes

import org.junit.jupiter.api.Test

class DuckNumberTest : AbstractNumberTest() {
	@Test
	fun testDuckNumbers() {
		testNumbersTypeByFile(com.local.numbertypes.NumberType.DUCK_NUMBER, "ducknumbers.txt")
	}

	@Test
	fun testDuckNumbersByList() {
		// Pass the order as 5
		testNumbersTypeByList(com.local.numbertypes.NumberType.DUCK_NUMBER, java.util.List.of<Int>(10, 20, 30, 40, 50, 60, 70, 80, 299593), 8)
	}
}